paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java

```diff
@@ -170,7 +171,8 @@ public List<DataField> parseSchemaChange() {
                     MySqlTypeUtils.toDataType(
                             column.get("typeName").asText(),
                             length == null ? null : length.asInt(),
-                            scale == null ? null : scale.asInt());
+                            scale == null ? null : scale.asInt(),
+                            MYSQL_CONVERTER_TINYINT1_BOOL.defaultValue());
```

Comment:
I think we should also pass this value to parser instead of using default value.
I think in schema evolution case, the TINYINT(1) should also obey the option.
You should add a test for the schema evolution test case.